cmd/gke-gcloud-auth-plugin/gcloud_token_provider.go

```diff
@@ -23,13 +23,26 @@ type gcloudConfiguration struct {
 
 // gcloudTokenProvider provides gcloud OAth 2.0 tokens.
 type gcloudTokenProvider struct {
-	readGcloudConfigRaw func() ([]byte, error)
-	readFile            func(filename string) ([]byte, error)
+	readGcloudConfigRaw       func() ([]byte, error)
+	readFile                  func(filename string) ([]byte, error)
+	account                   string
+	project                   string
+	impersonateServiceAccount bool
 }
 
 // readGcloudConfig returns an object which represents gcloud config output
 func (p *gcloudTokenProvider) readGcloudConfig() (*gcloudConfiguration, error) {
-	gcloudConfigBytes, err := p.readGcloudConfigRaw()
+	extraArgs := []string{}
+	if account != "" {
+		extraArgs = append(extraArgs, fmt.Sprintf("--account=%s", account))
+	}
+	if project != "" {
+		extraArgs = append(extraArgs, fmt.Sprintf("--project=%s", project))
+	}
+	if impersonateServiceAccount {
+		extraArgs = append(extraArgs, "--impersonate-service-account")
+	}
```

Comment:
Adding these fields will break the cache (sorry, when these fields were added, I was not added to the previous code reviews). So, we have to do a minor fix.

We should modularize "extraArgs" and Add a field to "type cache struct" calling it "GcloudExtraArgs string `json:"gcloud_extra_args"`"  

We should set this field in "writeGcloudAccessTokenToCache" and verify if this has changed  "getCachedGcloudAccessToken" functions. This will ensure the cache knows when args to gcloud are getting changed, and will invalidate the cache.